homeassistant/components/airtouch5/climate.py

```diff
@@ -0,0 +1,396 @@
+) -> None:
+    """Set up the Airtouch 5 Climate entities."""
+    client: Airtouch5SimpleClient = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities: list[ClimateEntity] = []
+
+    # Add each AC (and remember what zones they apply to)
+    zone_to_ac: dict[int, AcAbility] = {}
+    for ac in client.ac:
+        for i in range(ac.start_zone_number, ac.start_zone_number + ac.zone_count):
+            zone_to_ac[i] = ac
+        entities.append(Airtouch5AC(client, ac))
+
+    # Add each zone
+    for zone in client.zones:
+        entities.append(Airtouch5Zone(client, zone, zone_to_ac[zone.zone_number]))
+
+    async_add_entities(entities)
+
+
+class Airtouch5ClimateEntity(ClimateEntity, Airtouch5Entity):
+    """Base class for Airtouch5 Climate Entities."""
+
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_target_temperature_step = 1
+    _attr_name = None
+
+
+AC_MODE_TO_HVAC_MODE = {
+    AcMode.AUTO: HVACMode.AUTO,
+    AcMode.AUTO_COOL: HVACMode.AUTO,
+    AcMode.AUTO_HEAT: HVACMode.AUTO,
+    AcMode.COOL: HVACMode.COOL,
+    AcMode.DRY: HVACMode.DRY,
+    AcMode.FAN: HVACMode.FAN_ONLY,
+    AcMode.HEAT: HVACMode.HEAT,
+}
+HVAC_MODE_TO_SET_AC_MODE = {
+    HVACMode.AUTO: SetAcMode.SET_TO_AUTO,
+    HVACMode.COOL: SetAcMode.SET_TO_COOL,
+    HVACMode.DRY: SetAcMode.SET_TO_DRY,
+    HVACMode.FAN_ONLY: SetAcMode.SET_TO_FAN,
+    HVACMode.HEAT: SetAcMode.SET_TO_HEAT,
+}
+
+
+AC_FAN_SPEED_TO_FAN_SPEED = {
+    AcFanSpeed.AUTO: FAN_AUTO,
+    AcFanSpeed.QUIET: FAN_DIFFUSE,
+    AcFanSpeed.LOW: FAN_LOW,
+    AcFanSpeed.MEDIUM: FAN_MEDIUM,
+    AcFanSpeed.HIGH: FAN_HIGH,
+    AcFanSpeed.POWERFUL: FAN_FOCUS,
+    AcFanSpeed.TURBO: FAN_TURBO,
+    AcFanSpeed.INTELLIGENT_AUTO_1: FAN_INTELLIGENT_AUTO,
+    AcFanSpeed.INTELLIGENT_AUTO_2: FAN_INTELLIGENT_AUTO,
+    AcFanSpeed.INTELLIGENT_AUTO_3: FAN_INTELLIGENT_AUTO,
+    AcFanSpeed.INTELLIGENT_AUTO_4: FAN_INTELLIGENT_AUTO,
+    AcFanSpeed.INTELLIGENT_AUTO_5: FAN_INTELLIGENT_AUTO,
+    AcFanSpeed.INTELLIGENT_AUTO_6: FAN_INTELLIGENT_AUTO,
+}
+FAN_MODE_TO_SET_AC_FAN_SPEED = {
+    FAN_AUTO: SetAcFanSpeed.SET_TO_AUTO,
+    FAN_DIFFUSE: SetAcFanSpeed.SET_TO_QUIET,
+    FAN_LOW: SetAcFanSpeed.SET_TO_LOW,
+    FAN_MEDIUM: SetAcFanSpeed.SET_TO_MEDIUM,
+    FAN_HIGH: SetAcFanSpeed.SET_TO_HIGH,
+    FAN_FOCUS: SetAcFanSpeed.SET_TO_POWERFUL,
+    FAN_TURBO: SetAcFanSpeed.SET_TO_TURBO,
+    FAN_INTELLIGENT_AUTO: SetAcFanSpeed.SET_TO_INTELLIGENT_AUTO,
+}
+
+
+class Airtouch5AC(Airtouch5ClimateEntity):
+    """Representation of the AC unit. Used to control the overall HVAC Mode."""
+
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.FAN_MODE
+    )
+
+    def __init__(self, client: Airtouch5SimpleClient, ability: AcAbility) -> None:
+        """Initialise the Climate Entity."""
+        super().__init__(client)
+        self._ability = ability
+        self._attr_unique_id = f"ac_{ability.ac_number}"
```

Comment:
Is the ac number always the same?
Not sure if airtouch has an app, but the user has no option to change the order or something else...

Reply:
AC number should never change, unless physical changes have been done to the AC system (adding or removing a ducted AC unit).
This is not something that would normally happen.